trace/workflow_execution_update.go

```diff
@@ -0,0 +1,115 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+	"github.com/alitto/pond"
+	sdkclient "go.temporal.io/sdk/client"
+)
+
+type WorkflowExecutionUpdate struct {
+	State *WorkflowExecutionState
+}
+
+// WorkflowExecutionUpdateIterator is the interface the provides iterative updates, analogous to the HistoryEventIterator interface.
+type WorkflowExecutionUpdateIterator interface {
+	HasNext() bool
+	Next() (*WorkflowExecutionUpdate, error)
+}
+
+// WorkflowExecutionUpdateIteratorImpl implements the iterator interface. Keeps information about the last processed update and
+// receives new updates through the updateChan channel.
+type WorkflowExecutionUpdateIteratorImpl struct {
```

Comment:
Is the separation of interface and implementation for mocking purposes?

Reply:
Not really. I was just following the iterator pattern but didn't have a better name for the implementation. Happy to change this if it's not the right approach.

Reply:
Let's merge the interface and implementation.

Reply:
Not blocking but would be nice to address before merging.